### PR TITLE
Change response status code in case un-authorized request

### PR DESCRIPTION
**Pull request by ovekyc**
Change response status code in case un-authorized request

Response - 403

### CODE REVIEW

File: src/server/permission.js [hunk 1]
```diff
@@ -9,7 +9,13 @@ export const PERMISSION = {
 
 export const requiredPermission = (permission) => {
   return (req, res, next) => {
-    assert(req.headers.permission, 'No permission in request header');
+    if (!req.headers.permission) {
+      next(new APIError(new Error('No permission in request header'), {
+        statusCode: 403,
+        message: 'No permission in request header'
```
Comment (ochanje210) [inline src/server/permission.js:15]: shouldn't it be error instead of message key?

Comment (ovekyc) [reply src/server/permission.js:15]: No. again 
Please check out 
https://github.com/sv-bootcamp/goober/blob/master/src/server/ErrorHandler.js
and
https://github.com/sv-bootcamp/goober/blob/master/src/server/routes/api-routes.js#L33

Comment (ochanje210) [reply src/server/permission.js:15]: awesome. thanks

File: src/server/auth-token.js [hunk 2]
```diff
@@ -61,21 +61,22 @@ class AuthToken {
...
-        req.headers.permission = USER_PERMISSION[payload.userType];
-        next();
-      })
-      .catch(err => {
-        next(new APIError(err, {
-          statusCode: 400,
-          message: err.message
-        }));
-      });
+    AuthToken.decode(TOKEN_TYPE.ACCESS, jwtToken)
```
Comment (ochanje210) [inline src/server/auth-token.js:68]: why it is not returning now?

Comment (festinalente91) [reply src/server/auth-token.js:68]: return is not required. next() calls the next api function assigned in the router, but return can be used.

File: src/server/auth-token.js [hunk 3]
```diff
@@ -61,21 +61,22 @@ class AuthToken {
   static authenticate(req, res, next) {
     const bearerToken = req.headers.authorization;
     if (!bearerToken) {
-      return next();
+      next();
+      return;
```
Comment (ochanje210) [inline src/server/auth-token.js:65]: if return is not required when next is called, why do you have return here then?

File: src/server/auth-token.js (hunk 2 above)
Comment (ochanje210) [reply src/server/auth-token.js:68]: see the comment below

File: src/server/auth-token.js (hunk 3 above)
Comment (ovekyc) [reply src/server/auth-token.js:65]: This method does not have to return promise.
This method's goal is just call next() function after done its jobs.

But we need to keep consistency. I mean if it returns promise in some case and does not return promise (normal function) in some case, It is not consistent.

**Comment by festinalente91** [comment]
lgtm

**Comment by festinalente91** [comment]
lgtm

File: src/server/auth-token.js (hunk 2 above)
Comment (ovekyc) [reply src/server/auth-token.js:68]: resolve();

**Comment by kayla-kwon** [comment]
LCTM

**Comment by ovekyc** [comment]
LGTM

**Pull request merged**